core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1323,24 +1314,52 @@ private void commitIndexWriter(IndexWriter writer, Translog translog, String syn
         ensureCanFlush();
         try {
             Translog.TranslogGeneration translogGeneration = translog.getGeneration();
-            final Map<String, String> commitData = new HashMap<>(5);
 
-            commitData.put(Translog.TRANSLOG_GENERATION_KEY, Long.toString(translogGeneration.translogFileGeneration));
-            commitData.put(Translog.TRANSLOG_UUID_KEY, translogGeneration.translogUUID);
+            final String translogFileGen = Long.toString(translogGeneration.translogFileGeneration);
+            final String translogUUID = translogGeneration.translogUUID;
+            final String localCheckpoint = Long.toString(seqNoService().getLocalCheckpoint());
+            final String globalCheckpoint = Long.toString(seqNoService().getGlobalCheckpoint());
 
-            commitData.put(LOCAL_CHECKPOINT_KEY, Long.toString(seqNoService().getLocalCheckpoint()));
-            commitData.put(GLOBAL_CHECKPOINT_KEY, Long.toString(seqNoService().getGlobalCheckpoint()));
+            writer.setLiveCommitData(new Iterable<Map.Entry<String, String>>() {
+                // save the max seq no the first time its computed, so subsequent iterations don't recompute,
+                // potentially getting a different value
+                private String computedMaxSeqNoEntry = null;
```

Comment:
@bleskes I agree with using `lastCommittedSegmentInfos` from the engine as a solution, but my concern here was that basically we have to document and ensure that no one uses `IndexWriter#getLiveCommitData` or depends on it for accurate information, otherwise the max_seq_no could be different than what is actually stored in the commit.  That's why I added the caching part, which does increase complexity.  Do you prefer I remove it and just document that we should never call `IndexWriter#getLiveCommitData` inside ES code?